modules/ting/ting.module

```diff
@@ -468,20 +452,26 @@ function ting_ding_entity_buttons($type, $entity) {
       );
     }
 
-    $collection = ting_collection_load($entity->id);
-    if (count($collection->types) > 1) {
-      $buttons[] = array(
-        '#theme' => 'link',
-        '#text' => t('Other formats'),
-        '#path' => '#',
-        '#options' => array(
-          'attributes' => array(
-            'class' => array('action-button', 'other-formats'),
-          ),
-          'html' => FALSE,
-          'external' => TRUE,
-        ),
-      );
+    // Loading collection (specially on empty cache is expensive). So only load
+    // the collection when it's need and here only when this button is required
+    // in the design. Before using context it was load but not used in the
+    // design.
+    if (strpos($view_mode, 'teaser') === false) {
```

Comment:
Fixed and by looking through the design. It's very expensive to load the collection as this would also trigger loading of holdings. And the current design don't uses this in teaser views.

Reply:
My primary point is: We are adding a change in a "low level" module because of a specific "high level" design implementation.

This will add the button for all view modes that does not contain `teaser` - no matter whether they use the button or not.

Reply:
Come to think of it I think it would be appropriate to move the implementation to the companion `ding_ddbasic` module. The button does not do anything on it's own (path is `#`) and it requires [JavaScript currently available in the theme](https://github.com/ding2/ding2/blob/af8922fc3ff45e6344e07daa43cdf51b10ed5830/themes/ddbasic/scripts/ting-object/ting-object.js#L107) to do anything.